src/Swarm/TUI/Controller.hs

```diff
@@ -301,6 +306,12 @@ handleMainEvent ev = do
         do
           uiState . uiHideRobotsUntil .= t + TimeSpec 2 0
           invalidateCacheEntry WorldCache
+    -- debug focused robot
+    MetaChar 'd' | isPaused && hasDebug -> do
```

Comment:
It does not really work well with the game unpaused IMO. The focus jumps around too quickly and it is hard to make sense of it.

I will turn it off along with the single-step mode since they use the same toggle (at least for now).